nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/bulk/TestTrashAction.java

```diff
@@ -64,60 +59,35 @@
     @Inject
     public CoreSession session;
 
-    @Inject
-    public EventService eventService;
-
     @Inject
     public TransactionalFeature txFeature;
 
-    private static class TestListener extends EventListenerDescriptor {
-
-        private int count;
-
-        private TestListener() {
-            events = new HashSet<>(Collections.singleton(DOCUMENT_TRASHED));
-        }
-
-        @Override
-        public void initListener() {
-            inLineListener = event -> count++;
-        }
-    }
-
     @Test
     public void test() throws Exception {
 
         DocumentModel model = session.getDocument(new PathRef("/default-domain/workspaces/test"));
         String nxql = String.format("SELECT * from Document where ecm:ancestorId='%s'", model.getId());
 
-        TestListener listener = new TestListener();
-
-        eventService.addEventListener(listener);
+        try (CapturingEventListener listener = new CapturingEventListener(DOCUMENT_TRASHED)) {
+            String commandId = service.submit(
+                    new BulkCommand.Builder(ACTION_NAME, nxql).repository(session.getRepositoryName())
+                                                              .user(session.getPrincipal().getName())
+                                                              .param("value", Boolean.TRUE)
+                                                              .build());
 
-        String commandId = service.submit(
-                new BulkCommand.Builder(ACTION_NAME, nxql).repository(session.getRepositoryName())
-                                                          .user(session.getPrincipal().getName())
-                                                          .param("value", Boolean.TRUE)
-                                                          .build());
+            assertTrue("Bulk action didn't finish", service.await(Duration.ofSeconds(600)));
 
-        assertTrue("Bulk action didn't finish", service.await(Duration.ofSeconds(600)));
+            BulkStatus status = service.getStatus(commandId);
+            assertNotNull(status);
+            assertEquals(COMPLETED, status.getState());
+            assertEquals(CREATED_TOTAL, status.getProcessed());
 
```

Comment:
don't you need to eventService.waitForAsyncCompletion(); here ?

Reply:
No because the listener is not async.

Reply:
great then !